server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/DialectDiscoveryFolderService.java

```diff
@@ -76,29 +73,27 @@ public List<CobolDialect> loadDialects(CopybookService copybookService, MessageS
 
   /**
    * Load cobol dialects from the given path
-   * @param path is a path to the dialect
+   * @param uri is a URI to the dialect's jar file
    * @param copybookService a copybook service
    * @param messageService a message service
    * @return a list of loaded dialects
    */
   @Override
-  public List<CobolDialect> loadDialects(String path, CopybookService copybookService, MessageService messageService) {
+  public List<CobolDialect> loadDialects(URI uri, CopybookService copybookService, MessageService messageService) {
     try {
-      URI workdir = workingFolderService.getWorkingFolder(path);
-      return loadDialects(workdir, copybookService, messageService);
+      return createDialectsFromJar(uri, copybookService, messageService);
     } catch (Exception e) {
       warningCannotLoadDialects(e.getMessage());
     }
     return ImmutableList.of();
   }
 
-  private List<CobolDialect> loadDialects(URI workdir, CopybookService copybookService, MessageService messageService) {
+  private List<CobolDialect> loadDialectFromWorkingFolder(URI workdir, CopybookService copybookService, MessageService messageService) {
     try {
       return workingFolderService.getFilenames(workdir).stream()
           .filter(filename -> filename.startsWith("dialect-"))
           .filter(filename -> filename.endsWith(".jar"))
-          .map(filename -> createDialect(workdir, filename, copybookService, messageService))
-          .filter(Objects::nonNull)
+          .flatMap(filename -> createDialects(workdir, filename, copybookService, messageService).stream())
```

Comment:
Have a doubt. 
Is it possible to pass on the jar uri from the dialect client? In that case, we would be able to remove the jar naming constraint (dialect-*.jar).

Reply:
Here this method remains for loading dialects from the working folder after the start. If I remember correctly it is used for integration tests where we do not have dialect extensions

Reply:
Ahh understood. Thanks :)